pkg/virt-controller/watch/migration/migration_test.go

```diff
@@ -2264,25 +2262,208 @@ var _ = Describe("Migration watcher", func() {
+			Expect(priority).To(BeNumerically(">", migrationsutil.PriorityPending))
 			Expect(shutdown).To(BeFalse())
 			for i := range 5 {
 				if i == 3 {
 					continue
 				}
 				item, priority, shutdown := controller.Queue.GetWithPriority()
 				Expect(item).To(BeEquivalentTo(fmt.Sprintf("default/pending%d", i)))
-				Expect(priority).To(Equal(pendingPriority))
+				Expect(priority).To(Equal(migrationsutil.PriorityPending))
 				Expect(shutdown).To(BeFalse())
 			}
 		})
 
+		Context("with MigrationPriorityQueue feature gate enabled", func() {
+			BeforeEach(func() {
+				setConfig(&virtv1.KubeVirtConfiguration{
+					DeveloperConfiguration: &virtv1.DeveloperConfiguration{
+						FeatureGates: []string{featuregate.MigrationPriorityQueue},
+					},
+				})
+			})
+
+			It("should properly re-enqueue pending migrations with the defined priority when no new migration can start", func() {
+				By("Creating 1 pending migration. It will be picked up by the call to Execute()")
+				vmi := newVirtualMachine("testvmipending", virtv1.Running)
+				migration := newMigration("testmigrationpending", vmi.Name, virtv1.MigrationPending)
+				migration.Spec.Priority = pointer.P(virtv1.PrioritySystemCritical)
+				addMigration(migration)
+				addVirtualMachineInstance(vmi)
+				addPod(newSourcePodForVirtualMachine(vmi))
+
+				const migrationNamePrefix = "testmigration"
+				By("Creating 5 running system maintenance migrations")
+				for i := range 5 {
+					vmi := newVirtualMachine(fmt.Sprintf("testvmi%d", i), virtv1.Running)
+					migration := newMigration(fmt.Sprintf(migrationNamePrefix+"%d", i), vmi.Name, virtv1.MigrationRunning)
+					migration.Spec.Priority = pointer.P(virtv1.PrioritySystemMaintenance)
+					pod := newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+					addMigration(migration)
+					addVirtualMachineInstance(vmi)
+					addPod(pod)
+				}
+
+				By("Executing the controller and expecting the pending migration to have the defined priority")
+				controller.Execute()
+				runningMigrationsFromQueue := make([]string, 0, 5)
+				for range 5 {
+					item, priority, shutdown := controller.Queue.GetWithPriority()
+					runningMigrationsFromQueue = append(runningMigrationsFromQueue, item)
+					Expect(priority).To(Equal(0))
+					Expect(shutdown).To(BeFalse())
+				}
+				Expect(runningMigrationsFromQueue).To(
+					ConsistOf(
+						"default/"+migrationNamePrefix+"0",
+						"default/"+migrationNamePrefix+"1",
+						"default/"+migrationNamePrefix+"2",
+						"default/"+migrationNamePrefix+"3",
+						"default/"+migrationNamePrefix+"4",
+					),
+				)
+				item, priority, shutdown := controller.Queue.GetWithPriority()
+				Expect(item).To(Equal("default/testmigrationpending"))
+				Expect(priority).To(Equal(migrationsutil.PrioritySystemCritical))
+				Expect(shutdown).To(BeFalse())
+			})
+
+			It("should get items in order based on priority", func() {
+
+				const (
+					runningMigNamePrefix       = "testmigration"
```

Comment:
supernit - I was suggesting you include the "%d" part in these strings as well btw but it's not important